citesphere/src/main/java/edu/asu/diging/citesphere/web/validation/CitationConceptValidator.java

```diff
@@ -0,0 +1,34 @@
+package edu.asu.diging.citesphere.web.validation;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+
+import edu.asu.diging.citesphere.core.model.IUser;
+import edu.asu.diging.citesphere.core.service.ICitationConceptManager;
+import edu.asu.diging.citesphere.web.forms.CitationConceptForm;
+
+public class CitationConceptValidator implements Validator {
+
+    @Autowired
+    private ICitationConceptManager conceptManager;
+    
+    @Override
+    public boolean supports(Class<?> paramClass) {
+        return CitationConceptForm.class.equals(paramClass);
+    }
+
+    @Override
+    public void validate(Object target, Errors errors) {
+        CitationConceptForm conceptForm = (CitationConceptForm) target;
+        IUser user = (IUser)SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        if(!errors.hasErrors() && conceptForm.getAccessMode().equals("edit") &&
+                conceptManager.getByUriAndOwner(conceptForm.getUri(), user) == null) {
```

Comment:
I don't think this covers the case that the user is not the owner of the concept in question. If the user trying to modify this object has created another concept with the same URI as this one, this would return true but the user should still not be allowed to edit the concept.